src/main/java/gov/nasa/cumulus/metadata/aggregator/MetadataFilesToEcho.java

```diff
@@ -557,6 +559,81 @@ public IsoGranule readIsoMendsMetadataFile(String s3Location, Document doc, XPat
 		return  ((IsoGranule) granule);
 	}
 
+	public IsoGranule readFootprintOrbitBBox( Document doc, XPath xpath) throws XPathExpressionException{
+		if(this.isoXMLSpatialTypeEnumHashSet.contains(MENDsIsoXMLSpatialTypeEnum.BBOX)) {
+			AdapterLogger.LogDebug(this.className + " Based on MENDsIsoXMLSpatialTypeEnum, processing BBOX");
+			if (xpath.evaluate(IsoMendsXPath.NORTH_BOUNDING_COORDINATE, doc) != "") {
+				setGranuleBoundingBox(
+						Double.parseDouble(xpath.evaluate(IsoMendsXPath.NORTH_BOUNDING_COORDINATE, doc)),
+						Double.parseDouble(xpath.evaluate(IsoMendsXPath.SOUTH_BOUNDING_COORDINATE, doc)),
+						Double.parseDouble(xpath.evaluate(IsoMendsXPath.EAST_BOUNDING_COORDINATE, doc)),
+						Double.parseDouble(xpath.evaluate(IsoMendsXPath.WEST_BOUNDING_COORDINATE, doc)));
+			}
+		}
+		/**
+		 * There shall be no more logic of "if there is orbit, then there shall be no footprint" anymore
+		 * this function will solely depends on cumulus collection config to determine the collection is
+		 * GEODETIC, CARTICIAN , ORBIT
+		 * if isoXMLSpatialTypeEnumHashSet contains FOOTPRINT, means, the collection is either GEODETIC or CARTICIAN
+		 */
+		if(this.isoXMLSpatialTypeEnumHashSet.contains(MENDsIsoXMLSpatialTypeEnum.FOOTPRINT)) {
+			AdapterLogger.LogDebug(this.className + " Based on MENDsIsoXMLSpatialTypeEnum, processing FOOTPRINT");
+			try {
+				((IsoGranule) granule).setPolygon(xpath.evaluate(IsoMendsXPath.POLYGON, doc));
+				String line = xpath.evaluate(IsoMendsXPath.LINE, doc);
+				if (line != null && !line.isEmpty()) {
+					granule.getGranuleCharacterSet().add(createGranuleCharacter(line,"line"));
+				}
+			} catch (XPathExpressionException e) {
+				// Ignore if unable to parse for footprint since it isn't required for ingest
+				AdapterLogger.LogWarning(this.className + " Not able to extract MENDS footprint: " + e);
+			}
+		}
+		if(this.isoXMLSpatialTypeEnumHashSet.contains(MENDsIsoXMLSpatialTypeEnum.ORBIT)) {
+			AdapterLogger.LogDebug(this.className + " Based on MENDsIsoXMLSpatialTypeEnum, processing ORBIT");
+			((IsoGranule) granule).setOrbit(xpath.evaluate(IsoMendsXPath.ORBIT, doc));
+		}
+		return ((IsoGranule) granule);
+	}
+
+	public boolean isOrbitExisting(String orbitStr) {
+		if(StringUtils.isEmpty(StringUtils.trim(orbitStr))) {
+			return false;
+		} else{
```

Comment:
Nitpick: I would move this code out of the `else` block since the `if` block contains a return. That will reduce nesting and make the code easier to read.

Reply:
agree.  Move the code within the else block out of the else.  The code becomes much more clear.